res/controllers/Reloop-Digital-Jockey-2-IE.scripts.js

```diff
@@ -0,0 +1,878 @@
+
+
+////////////////////////////////////////////////////////////////////////
+// Quick Effects                                                      //
+////////////////////////////////////////////////////////////////////////
+
+/* HIGH kill / QuickEffect enable button */
+
+RDJ2.HighKillQuickEffectButton = function(options) {
+    this.channelNr = options.channelNr;
+    components.Button.call(this, options);
+};
+RDJ2.HighKillQuickEffectButton.prototype = new components.Button({
+    type: components.Button.prototype.types.powerWindow,
+    unshift: function() {
+        this.disconnect();
+        this.group = "[EqualizerRack1_[Channel" + this.channelNr + "]_Effect1]";
+        this.inKey = "button_parameter3";
+        this.outKey = "button_parameter3";
+        this.connect();
+        this.trigger();
+    },
+    shift: function() {
+        this.disconnect();
+        this.group = "[QuickEffectRack1_[Channel" + this.channelNr + "]_Effect1]";
+        this.inKey = "enabled";
+        this.outKey = "enabled";
+        this.connect();
+        this.trigger();
+    },
+});
+
+
+////////////////////////////////////////////////////////////////////////
+// Library                                                            //
+////////////////////////////////////////////////////////////////////////
+
+/* Trax knob */
+
+RDJ2.TraxKnob = function(options) {
+    components.Encoder.call(this, options);
+};
+RDJ2.TraxKnob.prototype = new components.Encoder({
+    group: "[Library]",
+    unshift: function() {
+        this.inKey = "MoveVertical";
+    },
+    shift: function() {
+        this.inKey = "ScrollVertical";
+    },
+    input: function(channel, control, value) {
+        var knobDelta = RDJ2.getKnobDelta(value);
+        this.inSetValue(knobDelta);
+    }
+});
+
+/* Trax button */
+
+RDJ2.TraxButton = function(obj) {
+    this.detectedDecks = [];
+    /* group and/or outKey cannot be defined at prototype initialization
+       (inside anonymous object passed to components.Button constructor
+       below) as this would cause additional premature engine.makeConnection
+       call that binds prototype object's output callback to the outKey.
+       This would cause uncaught exception when invoking the callback:
+
+       "TypeError: Result of expression 'this.detectedDecks' [undefined] is not an object"
+
+       which happens because at the time we create the prototype object,
+       we have no information yet about detectedDecks, as detectedDecks is
+       created at new TraxButton object construction, when its prototype is
+       long time existing. */
+    this.group = "[Master]";
+    this.outKey = "maximize_library";
+
+    this.detectDecks(obj);
+    components.Button.call(this);
+};
+RDJ2.TraxButton.prototype = new components.Button({
+    unshift: function() {
+        this.type = components.Button.prototype.types.toggle;
+        this.group = "[Master]";
+        this.inKey = "maximize_library";
+    },
```

Comment:
But it should toggle maximized view of library like it currently does. Same behaviour as in Traktor. This is the purpose of this button on this controller. It's even printed on it as a "PUSH2BROWSE" label. 

Why do you suggest to change this behaviour? Reading the description for "[Library]GoToItem", it's a completely different behaviour. I would stay with the simple toggle maximized view functionality.

Reply:
First of all, this is not Traktor, and we shouldn't copy any behaviour if it doesn't make sense in Mixxx. See https://github.com/mixxxdj/mixxx/wiki/Contributing-Mappings#design-guidelines
IMO "PUSH2BROWSE" is not very distinct. Push, hold & turn to browse?

`GoToItem` is a shortcut for actions you can usually not easily map on controllers, like expanding sidebar items, invoking any other action for selected tracks (see Preferences > Library > Double-Click Action), clear the search -- these are equally important but not accessible with your mapping.. 
I see you have `MoveFocusForward` as shift action. I suggest to use `GoToItem` for either shifted or unshifted press, and put `maximize_library` on the Trax longpress action for example, or push both Load buttons maybe?
Or use push-and turn for moving the focus: cw for `MoveFocusForward, ccw for `..Backward`?
What do you think?

Reply:
Thanks for the ideas. I will try to test them in a couple of days if you think it is worth the effort, as currently I am short on time.

Reply:
I have updated the mapping so that it contains `[Library],GoToItem` now. I have tested several configurations and I think it works best when it is mapped to `CLOSE/OPEN FOLDER` buttons (the shifted pitch bend buttons). It's the most compatible with the labels on the controller and most intuitive. And this compatibility matters, as I was really not sure what was the previous behaviour of the shifted pitch bend buttons. Pressing them caused vaguely noticable audible reaction and no GUI response as GUI responds only to `rate_temp_{up/down}` and not `rate_temp_{up/down}_small`. I had to investigate the mapping code to find out if something is wrong or not.

Now the library is browseable without the need for keyboard or mouse. Thanks again for the idea.